lib/picture.cpp

```diff
@@ -63,6 +63,37 @@ void Picture::save(const std::string& filename)
     fclose(fd);
 }
 
+std::vector<uint8_t> Picture::exportEncoded()
+{
+    auto png_ptr  = png_create_write_struct(PNG_LIBPNG_VER_STRING, nullptr, nullptr, nullptr);
+    auto info_ptr = png_create_info_struct(png_ptr);
+
+    std::vector<uint8_t> encoded;
+    // This callback will be called each time libpng wants to write an encoded chunk.
+    // https://github.com/Prior99/node-libpng/blob/master/native/encode.cpp
+    png_set_write_fn(png_ptr, &encoded, [] (png_structp png_ptr, png_bytep data, png_size_t length) {
+        auto encoded = reinterpret_cast<std::vector<uint8_t>*>(png_get_io_ptr(png_ptr));
+        encoded->insert(encoded->end(), data, data + length);
+    }, nullptr);
+
```

Comment:
Make sure to run clang-format on the files you change (if your IDE doesn't do it automatically). 
e.g. `clang-format -i ./lib/picture.cpp` it should pick up the style automatically